data/migrations/35.lua

```diff
@@ -1,3 +1,42 @@
 function onUpdateDatabase()
-	return false
+	print("> Updating database to version 36 (blessings update)")
+
+	-- skip BLESSING_ADVENTURERS_BLESSING & BLESSING_TWIST_OF_FATE
+	-- start at 10003001 + BLESSING_WISDOM_OF_SOLITUDE (2)
+	local storageRange = 10003003
+
+	local resultId = db.storeQuery("SELECT `id`, `blessings` FROM `players`")
+	if resultId then
+		local stmt = "INSERT INTO `player_storage` (`player_id`, `key`, `value`) VALUES "
+		repeat
+			local blessings = result.getNumber(resultId, "blessings")
+			local playerId = result.getNumber(resultId, "id")
+			for id = 0, 4 do
+				if bit.band(blessings, 2 ^ id) ~= 0 then
+					stmt = stmt .. "(" .. playerId .. ", " .. (storageRange + id) .. ", 1),"
+				end
+			end
+		until not result.next(resultId)
+		result.free(resultId)
+
+		local stmtLen = string.len(stmt)
+		if stmtLen > 66 then
+			stmt = string.sub(stmt, 1, stmtLen - 1)
+			db.query(stmt)
+		end
```

Comment:
Something looks wrong here - you're building one huge insert and then... trimming it to 66 chars?

Reply:
nope, only removing the last comma so the insert query doesn't fails

Reply:
66 it's the `stmt` length without any **inserted** values so if the `stmt` length is 66 we won't execute the insert query nor trim the string, but if it's greater than 66 it means that we have `rows` being inserted so we trim the `stmt` string and  remove the last comma

```sql
INSERT INTO `player_storage` (`player_id`, `key`, `value`) VALUES -- (66 length)
(1, 10003003, 1),
(1, 10003004, 1),
```
we'd have to remove the last comma so the query can be executed.
```sql
INSERT INTO `player_storage` (`player_id`, `key`, `value`) VALUES -- (66 length)
(1, 10003003, 1),
(1, 10003004, 1)
```

we could make an array for rows being inserted and concatenate them

```lua
local inserts = {}

inserts[#inserts + 1] = "..."

...

local stmt = "INSERT INTO ... VALUES "
if #inserts > 0 then
	stmt = stmt .. table.concat(inserts, ",")
	db.query(stmt)
end
```

Reply:
That would be good. Or even better, read all player ids and blessings first into a table, then you can skip building the query if there are no results i.e.

```lua
		rows = {}
		repeat
			local playerId = result.getNumber(resultId, "id")
			local blessings = result.getNumber(resultId, "blessings")
			for id = 0, 4 do
				if bit.band(blessings, 2 ^ id) ~= 0 then
					rows[#rows + 1] = {playerId, id}
				end
			end
		until not result.next(resultId)
```

or whichever structure you prefer :)